### PR TITLE
feat: add support for OpenSearch backend

### PR DESCRIPTION
**Pull request by maximepln**
Everything is in the title but I will add a bit of details

The goal of that PR is to implement a new backend provider `Opensearch`

For instance, I opened an issue about this there: [Issue 346](https://github.com/google/slo-generator/issues/346),
in which I explained why the `Elasticsearch` wasn't usable with `Opensearch`

As you will see, some code could have been mutualised between the two backend. But as we don't know the direction that will take the 2 backends in the future, and due to the fact that they could become quite different, I chose to avoid adding too much complexity. 

Furthermore, most of the people will work on either `Opensearch` or `Elasticsearch` so having a common codebase could become complicated when wanting to add new features.

Everything is documented in the right directories and unit tests & mocks were implemented.

### CODE REVIEW

**Comment by maximepln** [comment]
Hi @lvaylet, @bkamin29 

What are your thought on this ? 

I had to disable py-lint code duplication to have the CI to pass.

**Comment by lvaylet** [comment]
Hi @maximepln, thanks a lot for this submission. I will take a look at it shortly. Regarding your choice of going with two different backends (with some duplicate code), I ended up doing the exact same thing to support MQL in Cloud Monitoring on top of the existing MQF backend.

File: docs/providers/opensearch.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# Elasticsearch
+
+## Backend
+
+Using the `opensearch` backend class, you can query any metrics available in Opensearch to create an SLO.
```
Comment (lvaylet) [inline docs/providers/opensearch.md:5]: OpenSearch with a capital S, as used on [their website](https://opensearch.org/)? A global search/replace is probably the best option here.

File: slo_generator/backends/opensearch.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+from opensearchpy import OpenSearch
+
+from slo_generator.constants import NO_DATA
+
+LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=duplicate-code
+class OpensearchBackend:
```
Comment (lvaylet) [inline slo_generator/backends/opensearch.py:17]: Again, OpenSearchBackend with a capital S?

**Review by lvaylet — CHANGES_REQUESTED** [review]
A few minor changes left. Thanks for the good work!

**Comment by maximepln** [comment]
@lvaylet Thanks for the feedback,

I did the modification in the code. 

I voluntarily let the backend name to `opensearch` in the yaml file as they are all in snake_case and I didn't want to add any complexity on that side.

Don't hesitate if you have any other comment.

**Comment by maximepln** [comment]
Thanks for the feedback @lvaylet, 

I updated to code to use `OpenSearch` instead of `Opensearch`

Don't hesitate if you have any other comments

File: docs/providers/opensearch.md [hunk 3]
```diff
@@ -1,32 +1,32 @@
-# Elasticsearch
+# OpenSearh
 
 ## Backend
 
-Using the `opensearch` backend class, you can query any metrics available in Opensearch to create an SLO.
+Using the `OpenSearch` backend class, you can query any metrics available in OpenSearchBackend to create an SLO.
```
Comment (lvaylet) [inline docs/providers/opensearch.md:5]: available in **OpenSearch**

File: docs/providers/opensearch.md [hunk 4]
```diff
@@ -1,32 +1,32 @@
...
 backends:
-  opensearch:
+  open_search:
     url:
       - https://localhost:9200
       - https://localhost:9201
 ```
 
-The following method is available to compute SLOs with the `opensearch` backend:
+The following method is available to compute SLOs with the `OpenSearch` backend:
```
Comment (lvaylet) [inline docs/providers/opensearch.md:29]: `open_search` to remain consistent with the Cloud Monitoring documentation

**Review by lvaylet — CHANGES_REQUESTED** [review]
Thanks a lot for going over every instance of OpenSearch in the commits. I suggested a couple more changes to remain consistent with other backends (Cloud Monitoring specifically).

**Comment by maximepln** [comment]
Hi @lvaylet, 

Sorry for the delay to respond, I was away on holiday. 

Hope everything is fine now

**Comment by lvaylet** [comment]
I will take a look tomorrow afternoon.

**Review by lvaylet — APPROVED** [review]

**Pull request merged**